generation/internal/java.go

```diff
@@ -0,0 +1,82 @@
+package internal
+
+import (
+	"fmt"
+	"github.com/CircleCI-Public/circleci-config/config"
+	"github.com/CircleCI-Public/circleci-config/labeling/labels"
+)
+
+// latest LTS version
+const javaDockerImage = "cimg/openjdk:17.0"
+
+func javaTestJob(ls labels.LabelSet) *Job {
+	var cacheKey string
+	var cachePath string
+	var testCommand string
+	var testResultsPath string
+	var testReportsPath string
+
+	if ls[labels.ToolGradle].Valid {
+		gradleBuildFile := "build.gradle"
+		if ls[labels.FileBuildGradleKts].Valid {
+			gradleBuildFile = "build.gradle.kts"
+		}
+		cacheKey = fmt.Sprintf(`gradle-{{ checksum "%s" }}-{{ checksum "gradlew" }}`, gradleBuildFile)
+		cachePath = "~/.gradle/caches"
+		testCommand = "gradlew check"
+		testResultsPath = "build/test-results"
+		testReportsPath = "build/reports"
+	} else {
+		cacheKey = `maven-{{ checksum "pom.xml" }}`
+		cachePath = "~/.m2/repository"
+		testCommand = "mvn verify"
+		testResultsPath = "target/surefire-reports"
+	}
+
+	steps := []config.Step{
+		checkoutStep(ls[labels.DepsJava]),
+		{
+			Type:     config.RestoreCache,
+			CacheKey: cacheKey,
```

Comment:
The orb generates a cache key by searching for all the build.gradle/pom.xml files.
Maybe I should do that, too. 🤔